src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestCallback.cs

```diff
@@ -404,5 +408,26 @@ private static void OnRequestError(WinHttpRequestState state, Interop.WinHttp.WI
                     break;
             }
         }
+
+        private static void ResetAuthRequestHeaders(WinHttpRequestState state)
```

Comment:
Note: I debated whether or not I should put this method here or in WinHttpAuthHelper. Since the second option would involve adding a static instance of ResetCookieRequestHeaders to this class, I decided to put it here. If you think it should be in WinHttpAuthHelper or even somewhere else, just let me know.

Reply:
I think leaving it in WinHttpRequestCallback is fine because it is all part of the callback-Redirect logic.